packages/effect/src/Effect.ts

```diff
@@ -5392,3 +5392,27 @@ export const Tag: <const Id extends string>(id: Id) => <Self, Type>() =>
     })
     return done
   }
+
+/**
+ * Creates an `AbortSignal`.
+ * @since 3.1.0
+ */
+export const makeAbortSignal: Effect<AbortSignal> = map(
+  tap(
+    sync(() => new AbortController()),
+    (controller) => onInterrupt(() => sync(() => controller.abort()))
+  ),
+  (_) => _.signal
+)
+
+/**
+ * Creates an `AbortSignal` that is automatically aborted when the Scope is closed.
+ * @since 3.1.0
+ */
+export const makeAbortSignalScoped: Effect<AbortSignal, never, Scope.Scope> = map(
```

Comment:
Hmm will think about it. There aren't many cases where this is useful - usually the signals passed to tryPromise / async are all you need.

Reply:
Typically async and  promise are enough. But there have been a few circumstances where I've needed a signal inside of an Effect.gen for example and using async/promise didn't make sense.